kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -447,14 +450,22 @@ define([
             // 13 = enter, 27 = escape
             // the key is sometimes a string, hence == instead of ===
             if (e.key == 13 || e.key == 27) {
-                $(modalDialogNode).modal('hide');
                 if (e.key == 13 && args.type === 'confirm') {
                     resolution = true;
                 }
+                $(modalDialogNode).modal('hide');
             }
         });
 
         return new Promise((resolve) => {
+            if (args.type === 'confirm' && onConfirmFn) {
+                $(modalDialogNode).on('hide.bs.modal', () => {
+                    if (resolution) {
+                        resolution = onConfirmFn(resolution, modalDialogNode);
+                    }
```

Comment:
when the 'OK' button is clicked, run this function and put the results into the `resolution` variable, which is output when the modal completely disappears and resolves the promise.